src/Ocelot.Provider.Polly/PollyQoSResiliencePipelineProvider.cs

```diff
@@ -47,12 +49,15 @@ public ResiliencePipeline<HttpResponseMessage> GetResiliencePipeline(DownstreamR
     private void PollyResiliencePipelineWrapperFactory(ResiliencePipelineBuilder<HttpResponseMessage> builder, DownstreamRoute route)
     {
         var options = route.QosOptions;
-
-        // Add TimeoutStrategy if TimeoutValue is not int.MaxValue and greater than 0
+        
         if (options.TimeoutValue != int.MaxValue && options.TimeoutValue > 0)
         {
             builder.AddTimeout(TimeSpan.FromMilliseconds(options.TimeoutValue));
         }
+        else
+        {
+            builder.AddTimeout(TimeSpan.FromMilliseconds(route.Timeout));
```

Comment:
I still don't agree: route.timeOut is basically used to indicate the HttpClient timeout, which is different from Polly's own timeOut.
This means we can combine the 2.
As it is, I couldn't use this version.

Reply:
Rest assured, it won't be merged without your approval. 😄 
You're correct, @RaynaldM! We should not conflate the non-QoS `Timeout` with the QoS `TimeoutValue`.
Thus, if the `TimeoutValue` is defined, we should use it. If not, we should default to 90 seconds or a global value exclusively for QoS.

Reply:
@hogwartsdeveloper this `else`-block must read the global value from
https://github.com/ThreeMammals/Ocelot/blob/cc8f5c5dfc7c56e4446a2e80d97391429c8b364d/src/Ocelot/Configuration/File/FileGlobalConfiguration.cs#L24
Don't forget about absolute default value which must be redefined in this provider.